core/src/main/java/org/apache/carbondata/core/scan/filter/partition/EqualToFilterImpl.java

```diff
@@ -50,7 +50,7 @@ public EqualToFilterImpl(EqualToExpression equalTo, PartitionInfo partitionInfo)
           literal.getLiteralExpValue().toString(),
           partitionInfo.getColumnSchemaList().get(0).getDataType());
       if (PartitionType.RANGE == partitionInfo.getPartitionType() && value instanceof String) {
-        value = ByteUtil.toBytes((String)value);
+        value = ByteUtil.toBytesForPlainValue((String)value);
```

Comment:
It seems old overloaded method looks good than this name. It is utility method so better have a overloaded method

Reply:
`ByteUtil.toBytes` will do `val = val ^ Integer.MIN_VALUE;` 
So I think it can not be used as normal utility function to convert int to byte[4]